src/dump.c

```diff
@@ -529,6 +529,20 @@ static void jl_serialize_datatype(ios_t *s, jl_datatype_t *dt)
         tag = 4;
     else if (dt == jl_uint8_type)
         tag = 8;
+
+    if (strncmp(jl_symbol_name(dt->name->name), "#kw#", 4) == 0) {
+        /* XXX: yuck, but the auto-generated kw types from the serializer isn't a real type, so we *must* be very careful */
+        assert(tag == 0 || tag == 5 || tag == 6);
```

Comment:
are the meanings of these tags written down anywhere? should give them names instead of magic numbers